### ISSUE TITLE
false positive cppcoreguidelines-prefer-member-initializer warning, applying tweak produces broken code

### ISSUE THREAD

**Issue by geza-herman** [llvm/llvm-project#157734]
See this example:

```c++
struct Base {
    int a = 0;
};

template <typename T>
struct Der: Base {
    Der() {
        a = 1;
    }
};
```

clang-tidy produces `cppcoreguidelines-prefer-member-initializer` warning for `a = 1` in `Der`. Applying the suggested tweak (puts `a(1)` into `Der`'s member initializer list) creates broken code.

If `Der` is not a template, this problem doesn't happen.

**Comment by chrchr-github** [comment]
No repro with trunk: https://godbolt.org/z/89G8WMvEY
